### ISSUE TITLE
Erlang OTP version should below 23.0 to compile from source

### ISSUE THREAD

**Issue by haihongS** [ArweaveTeam/arweave#249]
rt.

**Comment by vkatsuba** [comment]
Hi @haihongS, please provide second information:
- What `Arweave` release you try build?
- From where you install `Erlang/OTP` 23.0?
- What operating system you use?
- Provide full stack trace of errors what you get during build of `Arweave`.

**Comment by haihongS** [comment]
Thanks @vkatsuba,

- master branch with id: e9aa880138e955cbdddcff4ecd3538c2d70a985e, 2.1.0.1
- install erlang/otp 23.0 from erlang installer
- macos 10.14.2
- x

I have fixed it right now, and several problems happened in my environment
### my env
- macos 10.14.2
- erlang/otp 23.0
- use erlang installer app in macos

### error
- due to the default erlang installer directory ```/Users/sherlock/Library/Application Support/ErlangInstaller```
the ```prometheus_process_collector``` can't compile correctly with space in directory name
- etc.

### fix
- add quote in makefile
- use brew to install erlang 22.3
- link directory without space

**Comment by vkatsuba** [comment]
Thanks @haihongS for deep details.
Looks like the error 
```
due to the default erlang installer directory /Users/sherlock/Library/Application Support/ErlangInstaller 
the prometheus_process_collector can't compile correctly with space in directory name
```
 was happened by reason that you was use directory name with white spaces. After put directory without white spaces in directory name, what other errors you get? Can you provide full log of errors?

**Comment by haihongS** [comment]
- I met the error ```library not found for -lerl_interface``` and found the similar problme in 
https://stackoverflow.com/questions/61947643/i-got-this-error-ld-library-not-found-for-lerl-interface

- white space quote: https://github.com/tsloughter/rebar3_run/pull/8

hope it will be helpful @vkatsuba

**Comment by haihongS** [comment]
cuz the problem has been solved, so it's hard for me to provide the full log of errors.
I checked the history of Chrome and found the problems which searched in Google

**Comment by vkatsuba** [comment]
Hi @ haihongS, many thanks for response. I apologize long updating, yesterday after open current issue we start investigate it and  we managed to reproduce current issue for Erlang release 23.0 by this reason we was create PR https://github.com/ArweaveTeam/arweave/pull/250 with some fixes. However, despite the fact that the assembly began to be successful we still have some issues with tests. Also, several third-party libraries what we use as dependency have the same issues, but after some research, we managed to find temporary solutions for these libraries what was provided by community, but all those fixes still waiting deep retesting and code review by contributors of libraries. For the moment, we will keep this ticket open until all proposed changes for libraries are reviewed/tested/applied by the library authors.
